core/store/src/db/rocksdb.rs

```diff
@@ -298,13 +297,11 @@ impl Database for RocksDB {
                 }
                 DBOp::DeleteAll { col } => {
                     let cf_handle = self.cf_handle(col)?;
-                    let opt_first = self.db.iterator_cf(cf_handle, IteratorMode::Start).next();
-                    let opt_last = self.db.iterator_cf(cf_handle, IteratorMode::End).next();
-                    assert_eq!(opt_first.is_some(), opt_last.is_some());
-                    if let (Some((min_key, _)), Some((max_key, _))) = (opt_first, opt_last) {
-                        batch.delete_range_cf(cf_handle, &min_key, &max_key);
+                    let range = self.get_cf_key_range(cf_handle).map_err(into_other)?;
```

Comment:
so if this returns 'Ok(None)' - we're not doing any deletions?

shoudl we print some warning / error then? Under what conditions does it happen?

Reply:
`None` means that the column is empty so it’s normal to not perform any deletions in that case.